server/service/devices.go

```diff
@@ -9,6 +9,40 @@ import (
 	"github.com/fleetdm/fleet/v4/server/fleet"
 )
 
+/////////////////////////////////////////////////////////////////////////////////
+// Fleet Desktop end points
+/////////////////////////////////////////////////////////////////////////////////
+type getFleetDesktopResponse struct {
+	Err             error `json:"error,omitempty"`
+	FailingPolicies uint  `json:"failing_policies_count"`
```

Comment:
@michalnicp asked:

> Should this also be omitempty? When there is an error, we should only return the error field in the json.

Uhmm, so if we just add `omitempty` to this, then the zero value will be omitted as well - which I don't think is correct because `{failing_policies_count:0}` is a valid response IMHO.

If we change this to a pointer, then we will need to deference this and add nil checks every-time we want to consume this - which over-complicates this.

But if that's the convection around the project, I'm onboard with the change.

Reply:
@lukeheath Is there a standard around this?

Reply:
Correct, status code is what we're really keying off of. The [convention](https://fleetdm.com/docs/using-fleet/rest-api#validation-failed2) for non-2xx is this format: 

```json
{
  "message": "Validation Failed",
  "errors": [
    {
      "name": "name",
      "reason": "cannot be empty"
    }
  ]
}
```